py4syn/epics/DxpClass.py

```diff
@@ -190,3 +215,70 @@ def close(self):
         """Stops an ongoing acquisition, if any, and puts the EPICS IOC in
         idle state."""
         self.pvDxpStop.put(1, wait=True)
+
+    def startCollectImage(self, rows=0, cols=0):
+        """Start to collect an image
+        When collect an image, the points will be  saved on a hdf file"""
+        self.rows = rows
+        self.cols = cols
+        # create HDF file
+        fileName = self.fileName
+        prefix = fileName.split('.')[0]
+
+        # TODO: include channel on fileName
+
+        fileName = self.fileName
+        idx = 1
+
+        while os.path.exists('%s_%s_%04d.hdf' % (prefix, self.dxpType, idx)):
+            idx += 1
+        fileName = '%s_%s_%04d.hdf' % (prefix, self.dxpType, idx)
+
+        self.fileResult = h5py.File(fileName)
+
+        # TODO: review this
+        lineShape = (1, self.rows, self.imageDeep)
+        # TODO: verify if it's better create it with complete or
+        # resize on each point
+        # TODO: verify if dtype is always int32
+        # create "image"
+        self.image = self.fileResult.create_dataset(
+                     'data',
+                     shape=(self.cols, self.rows, self.imageDeep),
+                     dtype='int32',
+                     chunks=lineShape)
+
+        # create "image" normalized
+        self.imageNorm = self.fileResult.create_dataset(
+                     'data_norm',
+                     shape=(self.cols, self.rows, self.imageDeep),
+                     dtype='float32',
+                     chunks=lineShape)
+
+        # last collected point
+        self.lastPos = 0
+
+    def stopCollectImage(self):
+        """Stop collect image"""
+        self.fileResult.close()
+        self.lastPos = -1
+
+    def setNormValue(self, value):
```

Comment:
The filename + saving to txt code appears in more than one place.
Would you mind to refactor it and make it usable for all the cases without repeating the code?